[CI] reindex oldEs090Fixture failure

Similar to https://github.com/elastic/elasticsearch/issues/51202, but occurring on the Linux builds. 

**Build scan**: https://gradle-enterprise.elastic.co/s/wvf3hs5bitzb2/console-log?anchor=1748

**Repro line**:

**Reproduces locally?**: didn't try

**Applicable branches**: master

**Failure history**: https://gradle-enterprise.elastic.co/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Execution%20failed%20for%20task%20*%0A%3E%20Failed%20to%20start%20*&search.relativeStartTime=P7D&search.timeZoneId=Europe/Berlin
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
**Log excerpt**:
```
oldEs090Fixture configuration:
-----------------------------------------
  cwd: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+platform-support-unix/os/sles-15&&immutable/modules/reindex/build/fixtures/oldEs090Fixture/cwd
  command: /var/lib/jenkins/.java/java11/bin/java oldes.OldElasticsearch /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+platform-support-unix/os/sles-15&&immutable/modules/reindex/build/fixtures/oldEs090Fixture /var/lib/jenkins/.gradle/caches/transforms-3/ca3951f17fb665f47250ce594d55743d/transformed/elasticsearch-0.90.13.zip true
  environment:
    JAVA_HOME: /var/lib/jenkins/.gradle/caches/transforms-3/3c259f9cc6236a487cd757f0d2f50ada/transformed/linux-8u302-x64.tar.gz
    CLASSPATH: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+platform-support-unix/os/sles-15&&immutable/test/fixtures/old-elasticsearch/build/distributions/old-elasticsearch-8.0.0-SNAPSHOT.jar:/var/lib/jenkins/.gradle/caches/modules-2/files-2.1/org.apache.lucene/lucene-core/9.0.0-snapshot-2719cf6630e/39c998e0de591232e0ca7e5c219e204369a185fc/lucene-core-9.0.0-snapshot-2719cf6630e.jar

  [run]
    "/var/lib/jenkins/.java/java11/bin/java" "$@" > run.log 2>&1 ; if [ $? != 0 ]; then touch run.failed; fi
oldEs090Fixture output:
-----------------------------------------
  failure marker exists: true
  pid file exists: false
  ports file exists: false

  [log]
    Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8
    Running [/var/lib/jenkins/.gradle/caches/transforms-3/ca3951f17fb665f47250ce594d55743d/transformed/elasticsearch-0.90.13.zip/elasticsearch-0.90.13/bin/elasticsearch, -f, -p, /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+platform-support-unix/os/sles-15&&immutable/modules/reindex/build/fixtures/oldEs090Fixture/pid]
    port not found
-----------------------------------------
:modules:reindex:oldEs090Fixture#stop SKIPPED
build complete, generating: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+platform-support-unix/os/sles-15&&immutable/build/348.tar.bz2

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':modules:reindex:oldEs090Fixture'.
> Failed to start oldEs090Fixture

```


**Failure excerpt**:
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':modules:reindex:oldEs090Fixture'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:174)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:402)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:389)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:382)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:368)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: org.gradle.api.GradleException: Failed to start oldEs090Fixture
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
	at org.elasticsearch.gradle.internal.test.AntFixture.fail(AntFixture.groovy:210)
	...
```

--- a/test/fixtures/old-elasticsearch/src/main/java/oldes/OldElasticsearch.java
+++ b/test/fixtures/old-elasticsearch/src/main/java/oldes/OldElasticsearch.java
@@ -92,7 +92,7 @@ public class OldElasticsearch {
             command.add("-f");
         }
         command.add("-p");
-        command.add(baseDir.resolve("pid").toString());
+        command.add("\"" + baseDir.resolve("pid") + "\"");
         ProcessBuilder subprocess = new ProcessBuilder(command);
         Process process = subprocess.start();
         System.out.println("Running " + command);

--- a/test/fixtures/old-elasticsearch/src/main/java/oldes/OldElasticsearch.java
+++ b/test/fixtures/old-elasticsearch/src/main/java/oldes/OldElasticsearch.java
@@ -92,7 +92,7 @@ public class OldElasticsearch {
             command.add("-f");
         }
         command.add("-p");
-        command.add("\"" + baseDir.resolve("pid") + "\"");
+        command.add(baseDir.resolve("pid").toString().replaceAll("&", "\\&"));
         ProcessBuilder subprocess = new ProcessBuilder(command);
         Process process = subprocess.start();
         System.out.println("Running " + command);

